### ISSUE TITLE
Search and select a suggestion in AutocompleteInput shows a wrong optionText.

### ISSUE THREAD

**Issue by devossa1** [marmelab/react-admin#3503]
**What you were expecting:**
When searching in an AutocompleteInput and selecting a suggestion, we expect to see the optionText of the selected item.

**What happened instead:**
When searching in an AutocompleteInput and selection a suggestion, in the text field we see the previously selected item's optionText.

**Steps to reproduce:**
1. Open the link below.
2. In the **test** AutocompleteInput, please, write "3" and select the suggestion. (The text field remains empty, does not show "test 3").
3. Now search for the "1". The value in the text fied becomes "test 3", not "test 1", as expected.

**Related code:**
CodeSandbox link: https://ykkri.csb.app/#/posts/create

We found a solution. In the **ra-ui-materialui/src/input/AutocompleteInput.js** file
```
handleSuggestionSelected = (event, { suggestion, method }) => {
    const { input } = this.props;

    const inputValue = this.getSuggestionValue(suggestion);
    if (input && input.onChange) {
      this.setState(
        {
           dirty: false,
          inputValue,//removing this also solves the problem
          //replacing "inputValue" with "searchText: inputValue," solves the problem, too
          selectedItem: suggestion
        },
        () => {
          input.onChange(inputValue);
        }
      );
    }

    if (method === "enter") {
      event.preventDefault();
    }
  };
```

**Other information:**

**Environment**

* React-admin version: 2.9.2
* Last version that did not exhibit the issue (if applicable):
* React version: 16.8.6
* Browser: Chrome 75.0.3770.142 
* Stack trace (in case of a JS error):

**Comment by djhi** [comment]
Thanks for the detailed issue. Confirmed

**Comment by devossa1** [comment]
Thanks

**Comment by Dogson** [comment]
Pulled my hair on that one. Thanks !

**Comment by JulienMattiussi** [comment]
This have been solved by #4367 

You now have to add also an inputText prop to your AutocompleteInput when the optionText is a Component

https://marmelab.com/react-admin/doc/3.2/Inputs.html#autocompleteinput
